python/utils/worker.py

```diff
@@ -92,195 +92,258 @@ def run(self):
-            stub = raw_data_pb2_grpc.RawDataStub(channel)
-            request = raw_data_pb2.GetTransactionsRequest(
-                starting_version=batch_start_version
-            )
-            responses = stub.GetTransactions(request, metadata=metadata)
-            responses = iter(responses)
+                assert response is not None
+                transactions = list(response.transactions)
 
-            while True:
-                perf_start_time = perf_counter()
-                transaction_batches: List[List[transaction_pb2.Transaction]] = []
+                current_batch_size = len(transactions)
+                if current_batch_size == 0:
+                    print(
+                        json.dumps(
+                            {
+                                "processor_name": self.processor.name(),
+                                "message": "[Parser] Received empty batch from GRPC stream",
+                                "starting_version": starting_version,
+                            }
+                        ),
+                        flush=True,
+                    )
+                    sys.exit(1)
 
-                # Get batches of transactions for processing
-                for _ in range(self.num_concurrent_processing_tasks):
-                    response = next(responses)
-                    transactions = response.transactions
+                batch_start_version = transactions[0].version
+                batch_end_version = transactions[-1].version
 
+                # If ending version is in the current batch, truncate the transactions in this batcch
+                if ending_version != None and batch_end_version >= ending_version:
+                    batch_end_version = ending_version
+                    transactions = transactions[
+                        : batch_end_version - batch_start_version + 1
+                    ]
                     current_batch_size = len(transactions)
-                    if current_batch_size == 0:
-                        print("[Parser] Received empty batch from GRPC stream")
-                        sys.exit(1)
 
-                    batch_start_version = transactions[0].version
-                    batch_end_version = transactions[-1].version
+                transaction_batches.append(transactions)
 
-                    # If ending version is in the current batch, truncate the transactions in this batcch
-                    if ending_version != None and batch_end_version >= ending_version:
-                        batch_end_version = ending_version
-                        transactions = transactions[
-                            : batch_end_version - batch_start_version + 1
-                        ]
-                        current_batch_size = len(transactions)
+                # If it is a partial batch, then skip polling and head to process it first.
+                if current_batch_size < INDEXER_GRPC_BLOB_STORAGE_SIZE:
```

Comment:
I don't think that we need to add special handling to partial batch anymore due to server side chunking.

Reply:
Or actually, can we switch to a similar model to rust where we pull in a separate thread?